Jenkinsfile

```diff
@@ -129,31 +131,68 @@ def getTests(String baseUrl) {
 }
 
 def getScenarios(String baseUrl, String testId){
-    def scenarios = [[Id: "-1", Name: "Default"]]
-    def isScenarioChoosen = "${params['Run with scenarios']}".toBoolean()
-    if (isScenarioChoosen) {
+    def scenarios = [[Id: "0", Name: "Default"]]
+    def isScenarioChoosen = "${params['Run with datasets']}".toBoolean()
+    def isScenarioPresentForTest = scenariosPresent(baseUrl, testId)
```

Comment:
What was the issue with the old approach? Making a request for scenarios for a test that doesn't have doesn't just return an empty array?

Reply:
yeah, it returns an empty array for the single dataset and for the test which doesn't have dataset at all.
and returns dataset for the test which has, but the question is does it use those dataset?

but the problem is when I create a test with multiple datasets suppose 3 (1 default and 2 from website I added) then when I remove the dataset parameter used in the test. it downloads all the dataset which are not required and run them, 3 times in our case.

useMango checks for Parameter in the test body to identify whether it needs to download dataset or not.
so, for example there is no parameter shown in the 
![image](https://github.com/infuse-consulting/jenkins/assets/60608318/b407d6d7-1693-4973-8538-0c5daf1ef625)
so useMango treats it as a normal test. and doesn't even show the dataset selection part.
even if there are dataset present.
![image](https://github.com/infuse-consulting/jenkins/assets/60608318/447563b3-0f73-44e2-af93-30cf661738aa)

All in all, for useMango default dataset is mandatory to identify whether it is a normal test or the dataset one.

i tried to replicate the same.

Reply:
Are you saying you can have datasets for a test that has no parameters? Do the datasets not get deleted when the last parameter is deleted?

Reply:
No, the one's which were created on the website still remains.